src/telemetry.ts

```diff
@@ -5,6 +5,19 @@
 import {DebugProtocol} from 'vscode-debugprotocol';
 import {OutputEvent} from 'vscode-debugadapter';
 
+export type ExceptionType = "uncaughtException" | "unhandledRejection" | "generic";
+export interface  IExecutionResultTelemetryProperties {
+    // There is an issue on some clients and reportEvent only currently accept strings properties,
+    // hence all the following properties must be strings.
+    successful?: "true" | "false";
+    exceptionType?: ExceptionType;
```

Comment:
Should this be called eventType or actionType rather than exceptionType?

Reply:
It is used to differentiate whether it's a uncaughtException or promiseRejection for exception case. Why should it be eventType or actionType?

Reply:
I think that people might expect to see the class name in a field called exceptionType.

Reply:
I think this word clearly conveys its meaning. Neither `eventType` or `actionType` is correct to capture the intention of this field.  The case you mention might be valid, but I feel that it might come from a limited group of people who might come from a scoped background. It is not that convincing that `exceptionType` has to mean a class name. Users of data should check the content of the data before make any assumptions. So unless we would have a absolutely perfect wording for this concept (other than actionType and eventType), I might leave it as-is.